Development/OpenPassDevelopmentApp/DeviceAuthorizationViewModel.swift

```diff
@@ -0,0 +1,126 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+#if os(tvOS)
+
+import CoreImage.CIFilterBuiltins
+import Foundation
+import OpenPass
+import SwiftUI
+
+extension DeviceAuthorizationViewModel {
+
+    /// A interface defining the flow of state communicated by the `DeviceAuthorizationFlow`
+    public enum State: Sendable {
+
+        /// The client has been initialized but a ``DeviceCode`` has not been requested or received
+        case initial
+
+        case deviceCodeAvailable(DeviceCode)
+
+        /// The previous ``DeviceCode`` has now expired, and the consumer is required to re-start the flow via
+        /// ``DeviceAuthorizationFlow.fetchDeviceCode()``.
+        case deviceCodeExpired
+
+        /// An unexpected error has occurred.
+        case error(Error)
+
+        /// The flow is complete and the associated ``OpenPassManager`` has obtained the set of ``OpenPassTokens``.
+        case complete(OpenPassTokens)
+    }
+}
+
+@MainActor
+final class DeviceAuthorizationViewModel: ObservableObject {
+
+    @Published private(set) var deviceCode: DeviceCode?
+
+    @Published private(set) var verificationUriCompleteImage: UIImage?
+
+    @Published private(set) var error: Error?
+
+    @Published private(set) var state: DeviceAuthorizationViewModel.State = .initial {
+        didSet {
+            switch state {
+            case .initial:
+                break
+            case .deviceCodeAvailable(let deviceCode):
+                self.deviceCode = deviceCode
+                if let verificationUriComplete = deviceCode.verificationUriComplete {
+                    verificationUriCompleteImage = qrCodeImage(url: verificationUriComplete)
+                }
+            case .deviceCodeExpired:
+                self.deviceCode = nil
+            case .error(let error):
+                self.error = error
+            case .complete:
+                break
+            }
+        }
+    }
+
+    private var signInTask: Task<Void, Never>?
+
+    public func startSignInDAFFlow() {
+        signInTask = Task {
```

Comment:
Nit: As good practice, in case this code is copied by publishers, would you want to call `cancelSignIn` at the beginning to ensure we don't have multiple `Task`s executing at the same time?